scripts/prepare_heat_demand.py

```diff
@@ -2,15 +2,439 @@
+
+    # Calculate mfh (mfh: Mehrfamilienhaus) heat load
+    heat_load_consumer["mfh" + "_" + carrier] = bdew.HeatBuilding(
+        heat_load_consumer.index,
+        holidays=holidays,
+        temperature=temperature,
+        shlp_type="MFH",
+        building_class=building_class,
+        wind_class=0,
+        annual_heat_demand=yearly_demands["mfh" + "_" + carrier][0],
+        name="MFH",
+        ww_incl=True,
+    ).get_bdew_profile()
+
+    # Calculate industry, trade, service (ghd: Gewerbe, Handel, Dienstleistung)
+    # heat load
+    heat_load_consumer["ghd" + "_" + carrier] = bdew.HeatBuilding(
+        heat_load_consumer.index,
+        holidays=holidays,
+        temperature=temperature,
+        shlp_type="ghd",
+        wind_class=0,
+        annual_heat_demand=yearly_demands["ghd" + "_" + carrier][0],
+        name="ghd",
+        ww_incl=True,
+    ).get_bdew_profile()
+
+    # Calculate total heat load in year
+    heat_load_total[carrier + "-demand-profile"] = heat_load_consumer.sum(axis=1)
+
+    # Normalize heat load profile
+    heat_load_total[carrier + "-demand-profile"] = np.divide(
+        heat_load_total[carrier + "-demand-profile"],
+        heat_load_total[carrier + "-demand-profile"].sum(),
+    )
+
+    return heat_load_total
+
+
+def postprocess_data(heat_load_postprocessed, heat_load_year, region, scenario, unit):
+    """
+    This function stacks time series of heat load profile and addes it to result DataFrame
+
+    Parameters
+    ----------
+    heat_load_postprocessed : pd.Dataframe
+        Empty Dataframe as result DataFrame
+    heat_load_year : pd.Dataframe
+        DataFrame with total normalized heat load in year to be processed
+    region : str
+        Region (eg. Brandenburg)
+    scenario : str
+        Scenario e.g. "base
+    unit : str
+        Unit of total demands (eg. GWh)
+
+    Returns
+    -------
+    heat_load_postprocessed : pd.DataFrame
+         DataFrame that contains stacked heat load profile
+
+    """
+    # Stack time series with total heat load in year
+    heat_load_year_stacked = dp.stack_timeseries(heat_load_year)
+
+    heat_load_year_stacked["region"] = region
+    heat_load_year_stacked["scenario"] = scenario
+    heat_load_year_stacked["var_unit"] = unit[0]
+
+    # Append stacked heat load of year to stacked time series with total heat loads
+    heat_load_postprocessed = pd.concat(
+        [heat_load_postprocessed, heat_load_year_stacked], ignore_index=True, sort=False
+    )
+
+    return heat_load_postprocessed
+
+
+if __name__ == "__main__":
+    in_path1 = sys.argv[1]  # path to weather data
+    in_path2 = sys.argv[2]  # path to household distributions data
+    in_path3 = sys.argv[3]  # path to holidays
+    in_path4 = sys.argv[4]  # path to building class
+    in_path5 = sys.argv[5]  # path to b3 scalars.csv
+    out_path = sys.argv[6]
+
+    REGION = ["BB", "B"]
+    SCENARIO = "base"
+    CARRIERS = ["heat_central", "heat_decentral"]
+
```

Comment:
I is preferable to load all the scalar data apart from the weather data before the loop to avoid loading the same data again and again. This makes some adaptions necessary: The respective functions should accept the data instead of a path.

Reply:
This is useful because we need to introduce another function that reads the scalar data, aggregates it and saves it in resources (as `scal_load_heat.csv` or similar).

Reply:
Does the changes I did in commit https://github.com/rl-institut/oemof-B3/pull/76/commits/9904d2861341d460138e29141276efb5953ae5a4 cover your suggestions? I loaded scalars before the loop and adapted the get_heat_demand function accordingly. I did not change the reading of the rest of the raw data (household distribution, holidays and building class).